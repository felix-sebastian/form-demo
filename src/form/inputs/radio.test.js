import React from "react";
import Radio from "./radio";
import { render, fireEvent, mockImplementation } from "@testing-library/react";

it("renders the active input", () => {
  const { baseElement } = render(
    <Radio options={["Foo", "Bar", "Baz"]} value={1} />
  );

  expect(baseElement).toMatchSnapshot();
});

it("calls handleChange when clicked", () => {
  let callback = jest.fn();

  const { getByText } = render(
    <Radio options={["Foo", "Bar", "Baz"]} value={1} handleChange={callback} />
  );

  fireEvent.click(getByText("Foo"));
  expect(callback).toHaveBeenCalledTimes(1);
});
