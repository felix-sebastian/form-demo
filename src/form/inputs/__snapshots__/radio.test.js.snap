// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the active input 1`] = `
<body>
  <div>
    <div
      class="sc-AykKE gxUNkH"
    >
      <div
        class="sc-AykKG ciIkEo"
        data-testid="row"
      >
        <div
          class="sc-AykKF iZvGwK"
          tabindex="0"
        />
        Foo
      </div>
      <div
        class="sc-AykKC iNGtUv"
      />
      <div
        class="sc-AykKG ciIkEo"
        data-testid="row"
      >
        <div
          class="sc-AykKF gBcjvM"
          tabindex="0"
        />
        Bar
      </div>
      <div
        class="sc-AykKC iNGtUv"
      />
      <div
        class="sc-AykKG ciIkEo"
        data-testid="row"
      >
        <div
          class="sc-AykKF iZvGwK"
          tabindex="0"
        />
        Baz
      </div>
    </div>
  </div>
</body>
`;
